sql_generators/active_users_versioned/templates/mobile_query.sql

```diff
@@ -0,0 +1,674 @@
+        first_seen_year,
+        is_default_browser,
+        language_name,
+        app_name,
+        channel,
+        os,
+        os_version,
+        os_version_major,
+        os_version_minor,
+        submission_date,
+        segment,
+        adjust_network,
+        install_source
+      ORDER BY
+        last_updated_timestamp
+    )
+),
+-- Next CTE returns is the historical data before adding the new column.
+_previous AS (
+  SELECT
+    * EXCEPT (
+      app_version_major,
+      app_version_minor,
+      app_version_patch_revision,
+      app_version_is_major_release,
+      os_grouped
+    ),
+    DATE_SUB(CURRENT_TIMESTAMP, INTERVAL 1 DAY) AS last_updated_timestamp
+  FROM
+    `{{ project_id }}.{{ app_name }}.active_users_aggregates` -- Query the view (not the table) in order to retrive records HAVING MAX(last_updated_timestamp)
+  WHERE
+    submission_date >= "2021-01-01"
+),
+-- Next CTE returns the union of _current and _previous, which is used later to calculate differences between historical and new data.
+-- NOTE. Using MERGE from here in query2 for better performance and clarity. It requires this file to run as a script.
+unioned AS (
+  SELECT
+    *
+  FROM
+    _previous
+  UNION ALL
+  SELECT
+    segment,
+    attribution_medium,
+    attribution_source,
+    attributed,
+    city,
+    country,
+    distribution_id,
+    first_seen_year,
+    is_default_browser,
+    channel,
+    os,
+    os_version,
+    os_version_major,
+    os_version_minor,
+    submission_date,
+    language_name,
+    MAX(dau) AS dau,
+    MAX(wau) AS wau,
+    MAX(mau) AS mau,
+    MAX(new_profiles) AS new_profiles,
+    MAX(ad_clicks) AS ad_clicks,
+    MAX(organic_search_count) AS organic_search_count,
+    MAX(search_count) AS search_count,
+    MAX(search_with_ads) AS search_with_ads,
+    MAX(uri_count) AS uri_count,
+    MAX(active_hours) AS active_hours,
+    adjust_network,
+    install_source,
+    app_name,
+    app_version,
+    last_updated_timestamp
+  FROM
+    _current
+  GROUP BY
+    app_version,
+    attribution_medium,
+    attribution_source,
+    attributed,
+    city,
+    country,
+    distribution_id,
+    first_seen_year,
+    is_default_browser,
+    language_name,
+    app_name,
+    channel,
+    os,
+    os_version,
+    os_version_major,
+    os_version_minor,
+    submission_date,
+    segment,
+    adjust_network,
+    install_source,
+    last_updated_timestamp
+),
+-- Next CTE returns the difference in metrics calculated between historical and new records for the new column. The delta is set as UNDETERMINED.
+backfill_delta AS (
+  SELECT
+    segment,
+    attribution_medium,
+    attribution_source,
+    attributed,
+    city,
+    country,
+    distribution_id,
+    first_seen_year,
+    is_default_browser,
+    channel,
+    os,
+    os_version,
+    os_version_major,
+    os_version_minor,
+    submission_date,
+    language_name,
+    IF(
+      last_updated_timestamp = @current_timestamp
```

Comment:
Check the usage of a query parameter or CURRENT_TIMESTAMP. This timestamp needs to match exactly to be able to make this comparison.